provider/provider_go_test.go

```diff
@@ -0,0 +1,33 @@
+//go:build !python && !yaml && !nodejs && !dotnet && !java
+// +build !python,!yaml,!nodejs,!dotnet,!java
+
+package provider
+
+import (
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestSnsTopicReplacement(t *testing.T) {
+	ptest := pulumiTest(t, filepath.Join("test-programs", "sns-topic-replacement"))
+	ptest.InstallStack("test")
+
+	topicName := randSeq(12)
+	ptest.SetConfig("topicName", topicName)
+
+	providerName := randSeq(12)
+	ptest.SetConfig("providerName", providerName)
+	ptest.Up()
+
+	// Change the name of the provider to check if topic replacement works.
+	providerName = randSeq(12)
```

Comment:
Does the test actually know it got replaced? Is there some small assert we could add to make it more obvious?

Reply:
yup, I've verified it from the logs but I'll also add an assert to make it more obvious